src/test/meson.build

```diff
@@ -757,7 +757,7 @@ tests += [
          [threads,
           libxz,
           liblz4],
-         '', '', '-DCATALOG_DIR="@0@"'.format(build_catalog_dir)],
+         '', '', '-DCATALOG_DIR="@0@"'.format(catalogdir)],
```

Comment:
I think we can drop
```
build_catalog_dir = meson.current_build_dir()
```
from catalog/meson.build as build_catalog_dir is no longer used anywhere in the code.

Reply:
Wouldn't this change make the test use the installed catalogs instead of the source ones?

Reply:
@fsateler CATALOG_DIR is only used when executed from the build directory.

> Wouldn't this change make the test use the installed catalogs instead of the source ones?

not afaics, the patch changes the logic by trying to detect if the binary was exectuted from the build directory and if so constructs the catalog dir via ```catalog_dir = path_join(NULL, ABS_BUILD_DIR, "catalog")```